code/modules/power/supermatter/supermatter.dm

```diff
@@ -291,10 +294,14 @@ GLOBAL_DATUM(main_supermatter_engine, /obj/machinery/power/supermatter_crystal)
 			SEND_SIGNAL(M, COMSIG_ADD_MOOD_EVENT, "delam", /datum/mood_event/delam)
 	if(combined_gas > MOLE_PENALTY_THRESHOLD)
 		investigate_log("has collapsed into a singularity.", INVESTIGATE_SUPERMATTER)
-		if(T)
-			var/obj/singularity/S = new(T)
-			S.energy = 800
-			S.consume(src)
+	if(T)
+		var/obj/singularity/S = null
+		if(ate_goose == TRUE)
+			S = new /obj/singularity/vomit(T)
+		else
+			S = new /obj/singularity(T)
+		S.energy = 800
+		S.consume(src)
```

Comment:
Indentation, if I am looking at this correctly it turns into a singularity always even when normally exploding.

Reply:
didn't spawn any singularity when i tried a standard delamination